client/src/components/NewIngredient.js

```diff
@@ -0,0 +1,194 @@
+    amount: "",
+    unit: "",
+    description: ""
+  })
+  const [errors, setErrors] = useState({})
+  const [serverErrors, setServerErrors] = useState([])
+  const [showOther, setShowOther] = useState(false)
+  const [shouldRedirect, setShouldRedirect] = useState(false)
+
+  const validateData = (formData) => {
+    setErrors({});
+    const { name, amount, unit } = formData
+    let foundError = false
+    let newErrors = {};
+    if (name.trim() == "") {
+      newErrors = {
+        ...newErrors,
+        name: "is required",
+      };
+      foundError = true
+    }
+
+    if (amount.trim() == "") {
+      newErrors = {
+        ...newErrors,
+        amount: "is required",
+      };
+      foundError = true
+    }
+
+
+    if (unit.trim() == "") {
+      newErrors = {
+        ...newErrors,
+        unit: "is required",
+      };
+      foundError = true
+    }
+
+    if (unit.trim() == "other") {
+      newErrors = {
+        ...newErrors,
+        unit: "please enter units"
+      }
+    }
+
+    setErrors(newErrors);
+    return foundError
+  }
+
+  const handleSubmit = async (event) => {
+    event.preventDefault()
+    const foundError = validateData(formData)
+    if (!foundError) {
+      const formPayload = {
+        newIngredient: { ...formData },
+        userId: props.user.id
+      }
+      try {
+        const response = await fetch(`/api/v1/ingredients`, {
+          method: "POST",
+          headers: new Headers({
+            "Content-Type": "application/json"
+          }),
+          body: JSON.stringify(formPayload)
+        })
+        if(!response.ok) {
+          if(response.status === 422) {
+            const body = await response.json()
+            const newErrors = translateServerErrors(body.errors)
+            return setServerErrors(newErrors)
+          } else {
+            const errorMessage = `${response.status}: (${response.statusText})`
+            const error = new Error(errorMessage)
+            throw(error)
+          }
+        } else {
+          setShouldRedirect(true)
+        }
+      } catch(error) {
+        console.error(`Error in Fetch: ${error.message}`)
+      }
+    }
+  }
+
+  const handleChange = (event) => {
+    setFormData({
+      ...formData,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+  }
+
+  const handleSelect = (event) => {
+    setFormData({
+      ...formData,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+    if (event.currentTarget.value === "other") {
+      setShowOther(true)
+    } else {
+      setShowOther(false)
+    }
+  }
+
+  let otherField
+  if (showOther) {
+    otherField = (
+      <div>
+        <label htmlFor="other">Other: </label>
+          <input 
+            type="text" 
+            id="other" 
+            name="unit"
+            value={formData.unit}
+            onChange={handleChange}
+            />
+      </div>
+    )
+  }
+
+  if (shouldRedirect) {
+    location.href="/ingredients"
```

Comment:
unless you need to re-retrieve information for your user state, you should prefer using a Redirect component for redirecting